06.bowling_object/lib/game.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  attr_reader :frames
+
+  def initialize(shot_pins)
+    @frames = build_frames(shot_pins)
+  end
+
+  def score
+    @frames.each_with_index.sum { |frame, idx| frame.score(@frames, idx) }
+  end
+
+  private
+
+  def build_frames(shot_pins)
+    @frames = []
+    shot_pins.split(',').each do |pin|
+      @frames = [*@frames, Frame.new] if frame_terminated?(@frames.last) && !last_frame?
```

Comment:
```suggestion
    shot_pins.split(',').each_with_index do |pin, i|
      @frames = [*@frames, Frame.new] if i.zero? || @frames.last.terminated? && !last_frame?
```

みたいにして、frame自身にterminatedかどうかを判定させるのがいいかもしれません。
ただ、frame.terminated? だと「フレームが終了した」みたいな意味になるので、ネーミング的に「？ 🤔 」かも

Reply:
frameクラスにterminatedを移動しました

ネーミングについて
「フレームが終了した」という意味にしようとしたのですが問題ありましたでしょうか？

.last.terminated? となるのが「最終の最終」のようにみえて読みづらかったので、framesの末尾であることを [-1] で表すように修正させていただきました。